resource/resource_graph.go

```diff
@@ -628,6 +644,11 @@ func (g *Graph) isNodeDependingOn(node, child Name) bool {
 func (g *Graph) SubGraphFrom(node Name) (*Graph, error) {
 	g.mu.Lock()
 	defer g.mu.Unlock()
+
+	return g.subGraphFrom(node)
+}
+
+func (g *Graph) subGraphFrom(node Name) (*Graph, error) {
```

Comment:
to confirm, did you mean to suggest renaming the public function to `SubGraphFromWithMutex`, or renaming this private function to `subGraphFromWith*out*Mutex` (asterisks for emphasis)

Reply:
hmm, for me that's a bit misleading since I would expect the function name to communicate what it is doing - this name assumes that is being called with a mutex held upstream.

Reply:
Given functions (by default) always take the locks they require, the confusion isn't registering on my side. I'm not going to hold the PR up over the naming. But there needs to be documentation that the caller must not be holding the graph mutex.

Reply:
yeah i was going to say i’m flexible on naming also as long as there is a doc-string that states the behavior. will add that before merging